fix: readiness check (#1666)
cap-js/docs: guides/deployment/health-checks.md

[Learn more about health checks on Kubernetes.](https://kubernetes.io/docs/tasks/configure-pod-container/configure-liveness-readiness-startup-probes) {.learn-more}

::: warning Limited Support on CF
Although supported by the Cloud Foundry core, readiness checks are not yet supported by the Cloud Foundtry CLI as well as the Cloud MTA Build Tool (MBT).
:::

> From `@sap/cds^7.8` onwards, the Node.js runtime provides an out-of-the-box endpoint for liveness and readiness checks at `/health`.

Author: no, i guess not :)

> From `@sap/cds^7.8` onwards, the Node.js runtime provides an out-of-the-box endpoint for liveness and readiness checks at `/health`.

Author: ```suggestion
The Node.js runtime provides an out-of-the-box endpoint for liveness and readiness checks at `/health`.
```

Requests that reach this public endpoint are answered with the status code 200 and the body `{ status: 'UP' }`.

> Since CAP Java v3.2, the `cds add` scripts `mta`, `cf-manifest`, `helm`, and `helm-unified-runtime` add the necessary dependencies and configuration for publicly available probe endpoints `/actuator/health/liveness` and `/actuator/health/readiness`.

Reviewer: ```suggestion
Since CAP Java v3.2, the `cds add` facets `mta`, `cf-manifest`, `helm`, and `helm-unified-runtime` add the necessary dependencies and configuration for the publicly available probe endpoints `/actuator/health/liveness` and `/actuator/health/readiness`.
```

> Since CAP Java v3.2, the `cds add` facets `mta`, `cf-manifest`, `helm`, and `helm-unified-runtime` add the necessary dependencies and configuration for the publicly available probe endpoints `/actuator/health/liveness` and `/actuator/health/readiness`.

Author: ```suggestion
For CAP Java, the `cds add` facets `mta`, `cf-manifest`, `helm`, and `helm-unified-runtime` add the necessary dependencies and configuration for the publicly available probe endpoints `/actuator/health/liveness` and `/actuator/health/readiness`.
```

For more details on adding Spring Boot health checks to your CAP Java app, please see [Spring Boot Health Checks](../../java/operating-applications/observability#spring-health-checks).

<!--

To achieve this, from `@sap/cds-dk^7.8` onwards, the configuration for readiness health checks is included in the MTA template for deployment to CF.
The default configuration specifies checks via http to `/` for Java (as there aren't any default Spring Boot health check endpoints) and `/health` for Node.js, respectively.

Additionally, for Node.js, the Helm chart template now specifies `/health` instead of `/` as the endpoint for the checks. For your app to be considered "alive", the endpoint `/health` needs to return a success response code.
This means, if you have a _fully custom_ `server.js`, you will need to add the `/health` endpoint to it, or adjust your MTA descriptor/Helm chart (if it's generated with `@sap/cds-dk^7.8`).

-->

> Completing the setup, `@sap/cds-dk` adds the necessary configurations to Helm charts for deployment to Kyma/ Kubernetes.

Reviewer: ```suggestion
For deployment to Kyma/ Kubernetes, `@sap/cds-dk` adds the necessary configurations to Helm charts to make the setup complete.
```
